### DISCUSSION TITLE
Some suggestions

### DISCUSSION THREAD

**Discussion by fluffynyako** [organicmaps/organicmaps#9436] [Ideas]
I have some suggestions for Organic Maps. I noticed that someone previously made a suggestion regarding road speed limits in the Telegram group chat, and this suggestion was adopted. Today, I discovered this feature while navigating, and I was pleasantly surprised. So, I would like to offer some suggestions as well, hoping they might be considered.

1. Provide more detailed voice navigation instructions. While some people may prefer concise navigation prompts, I believe most users would appreciate more detailed guidance. I recommended Organic Maps to a friend, and they mentioned that the navigation instructions were not detailed enough. Therefore, I hope you can offer more comprehensive navigation options. Sometimes, I also use OsmAnd because it provides more detailed voice navigation. My suggestions are as follows:
   a. Take a cue from OsmAnd and include various traffic alerts, such as stop signs, pedestrian crossings, speed bumps, tunnels, railroad crossings, toll booths, and more. These should not only be provided as voice alerts but also displayed on the screen. Users should be able to customize which types of traffic alerts they want and whether they prefer screen notifications, voice alerts, or both.
   b. Add prompts for continuing along the current road. My father usually uses OsmAnd while driving, and when he used Organic Maps on the Pan-American Highway, he found it too quiet. OsmAnd would remind him every few minutes to continue along the current road for a certain number of kilometers, and it would also alert him when overspeeding. But Organic Maps remained very silent, even when he had to keep driving straight for over 60 kilometers. My father found this quite boring, so I hope you can add a voice prompt that reminds users to continue along the current road for a certain distance, with an option to set the frequency of these prompts, allowing users to turn this feature on or off.
   c. Provide voice prompts for unpaved roads. Currently, there are warnings for unpaved roads, but they are limited to on-screen alerts without any voice prompts. Since there’s already an on-screen alert, I hope you can add a voice prompt to warn users when they’re about to enter an unpaved road, and allow users to enable or disable this feature.
   d. Allow users to select the frequency of turn prompts, such as “turn left in X meters.” Some users might prefer more frequent prompts, while others might like it simpler.
   e. When starting navigation, use voice prompts to announce the total distance and estimated time of arrival. This feature should also be optional for users to enable or disable.
   f. Include overspeed alerts. Although Organic Maps now has a speed limit feature, it still seems to lack overspeed alerts. Currently, it only supports speed camera alerts, but I hope you can add this feature as well, allowing users to manually enable or disable it.

2. Display more detailed information when users click on a POI (Point of Interest), and include toggles for this information.
   a. Add a photo feature that reads the “image” tag links from OpenStreetMap data. However, images should never load automatically; users must click to view them. I’m a bit puzzled by the current setting in Organic Maps, which mentions that more detailed information, like photos, requires mobile data. But does this only apply to Kayak? Why can’t it retrieve the “image” tag links? I hope this feature can be added.
   b. Include more detailed information, similar to OsmAnd, which displays additional details like place descriptions, religion and denomination, payment methods, and more, to make the map richer in detail.
   c. I would also like to see an option in the settings that allows users to choose which information to display or hide. For example, if I’m not disabled, I don’t need wheelchair accessibility information, so I can turn it off, making the map less cluttered.

3. Add support for more types of POIs, draw more icons, and display building types.
   a. I think Organic Maps needs to support more types of POIs. For example, I know of a radio station that is tagged as an amenity in Organic Maps, but it only shows the POI name without an icon. In OpenStreetMap, it’s tagged as amenity=studio and studio=radio, so I hope more POI types can be supported.
   b. I would like more icons to be drawn on the map. For instance, items like flagpoles and wind turbines are currently represented by just a small black dot on the map. For such POIs, I hope specific icons can be drawn.
   c. Display the type of building when users select a building. OpenStreetMap can tag different types of buildings, such as building=residential, building=commercial, building=retail, etc. However, regardless of the building type, when users select it in Organic Maps, it only shows a large title "Building." I hope the building type can be displayed, such as Residential Building, Commercial Building, Retail Building, etc. Only when it’s tagged as building=yes should it display as just "Building."

4. Add more options in the map styles and layers section.
   a. I think a "More Details" option could be added under map styles and layers. While there are currently options like outdoors, contour lines, and subway, I think a "More Details" option could be added that, when enabled, shows more details when users browse the map, such as street lights, speed bumps, pedestrian crossings, utility poles, traffic signs, traffic lights, etc. Some users might prefer a simpler map, while others, like me, prefer a more detailed map. So, I hope this feature can be added.
   b. I think a feature to "color buildings by type" could also be added. This would be similar to displaying building types but would use different colors for different types of buildings. However, the colors should not have too high contrast or saturation to avoid looking too garish; just enough to allow users to distinguish between them, balancing practicality with aesthetics.
   c. I also think a "Road Surface" feature could be added. This would be useful for people who enjoy activities like hiking. If they are biking or driving, it would be great if they could view information about the road surface! This could be done by using small colored lines or dashes on the road to indicate different surfaces, making it easier for users to distinguish road types.

5. Other Features
   a. A simulation navigation feature is something I’ve always wanted. I hope a simulation mode can be added, where after setting the starting point and destination, it simulates the navigation. Users should be able to choose the speed, with normal speed reflecting typical driving speed, based on speed limits and road types. Faster speeds could also be selected for a quicker simulation.
   b. A trip recording feature, which I know you are already developing, but I hope it can be completed soon, as this feature is really fantastic!

Finally, I want to thank all the developers and contributors of Organic Maps. Not only have you provided an excellent map navigation app, but you also helped me discover my passion. I only started getting to know and contributing to OpenStreetMap after discovering Organic Maps. Thank you for the guidance and help over the past year. I hope Organic Maps continues to improve! I also want to contribute to Organic Maps' development and code, or even fork Organic Maps, but unfortunately, I don’t know how to code. So, I can only contribute translations. I’ve made a few contributions to Organic Maps’ Chinese translations. If there’s ever a need for translation in the future, feel free to reach out to me. Chinese is my native language, and it’s very complex, so machine translations often make mistakes. Let’s work together to improve this excellent map navigation app!

**Comment by patepelo** [comment]
Great post @felixng1988  Thank you for keeping close to the project. Please understand that many of these ideas either exist or are better on an issue of its own. Let's tackle them step by step.

**Comment by biodranik** [comment]
Thanks for the detailed feedback, Felix. What is the value of repeating OSMand's functionality in Organic Maps?
